src/test/java/hudson/plugins/throttleconcurrents/ThrottleStepTest.java

```diff
@@ -0,0 +1,278 @@
+                Queue.Item i = story.j.jenkins.getQueue().getItems()[0];
+                assertTrue(i.task instanceof FreeStyleProject);
+
+                Node n = story.j.jenkins.getNode("first-agent");
+                assertNotNull(n);
+                assertEquals(1, n.toComputer().countBusy());
+                hasPlaceholderTaskForRun(n, firstJobFirstRun);
+                SemaphoreStep.success("wait-first-job/1", null);
+                story.j.assertBuildStatusSuccess(story.j.waitForCompletion(firstJobFirstRun));
+
+                FreeStyleBuild freeStyleBuild = futureBuild.waitForStart();
+                assertEquals(1, n.toComputer().countBusy());
+                for (Executor e : n.toComputer().getExecutors()) {
+                    if (e.isBusy()) {
+                        assertEquals(freeStyleBuild, e.getCurrentExecutable());
+                    }
+                }
+
+                WorkflowJob secondJob = story.j.jenkins.createProject(WorkflowJob.class, "second-job");
+                secondJob.setDefinition(getJobFlow("second", ONE_PER_NODE, "first-agent"));
+
+                WorkflowRun secondJobFirstRun = secondJob.scheduleBuild2(0).waitForStart();
+                story.j.waitForMessage("Still waiting to schedule task", secondJobFirstRun);
+                assertFalse(story.j.jenkins.getQueue().isEmpty());
+
+                assertEquals(1, n.toComputer().countBusy());
+                for (Executor e : n.toComputer().getExecutors()) {
+                    if (e.isBusy()) {
+                        assertEquals(freeStyleBuild, e.getCurrentExecutable());
+                    }
+                }
+                semaphore.release();
+
+                story.j.assertBuildStatusSuccess(story.j.waitForCompletion(freeStyleBuild));
+                SemaphoreStep.waitForStart("wait-second-job/1", secondJobFirstRun);
+                assertTrue(story.j.jenkins.getQueue().isEmpty());
+                assertEquals(1, n.toComputer().countBusy());
+                hasPlaceholderTaskForRun(n, secondJobFirstRun);
+                SemaphoreStep.success("wait-second-job/1", null);
+                story.j.assertBuildStatusSuccess(story.j.waitForCompletion(secondJobFirstRun));
+            }
+        });
+    }
+
+    private CpsFlowDefinition getJobFlow(String jobName, String category, String label) {
+        // This should be sandbox:true, but when I do that, I get org.jenkinsci.plugins.scriptsecurity.sandbox.RejectedAccessException: Scripts not permitted to use method groovy.lang.GroovyObject invokeMethod java.lang.String java.lang.Object
+        // And I cannot figure out why. So for now...
+        return new CpsFlowDefinition("throttle('" + category + "') {\n" +
+                "  echo 'hi there'\n" +
+                "  node('" + label + "') {\n" +
+                "    semaphore 'wait-" + jobName + "-job'\n" +
+                "  }\n" +
+                "}\n", false);
```

Comment:
Soooo - as I mentioned a few lines up, I'm getting script security exceptions on this when the sandbox is enabled and I can't figure out why, so for the moment, disabling script security until someone (probably @jglick!) tells me what I'm doing wrong to cause those errors to show up. =)